### DISCUSSION TITLE
Problem about Build Instructions

### DISCUSSION THREAD

**Discussion by Richard-happy** [libMesh/libmesh#3622] [Q&A]
I followed README.md to install libmesh. When I stepped in the stage of **6. make**, there is an error which shows that

Making all in .
make[1]: Entering directory '/home/LJL/libmesh/build'
  CXX      src/apps/fparser_parse_dbg-fparser_parse.o
In file included from ./include/libmesh/function_base.h:24,
                 from ./include/libmesh/parsed_function.h:22,
                 from ../src/apps/fparser_parse.C:23:
./include/libmesh/libmesh_common.h:27:4: error: #error DEBUG and NDEBUG should never be defined simultaneously
   27 | #  error DEBUG and NDEBUG should never be defined simultaneously
      |    ^~~~~
make[1]: *** [Makefile:32526: src/apps/fparser_parse_dbg-fparser_parse.o] Error 1
make[1]: Leaving directory '/home/LJL/libmesh/build'
make: *** [Makefile:33169: all-recursive] Error 1
 
Could you please help me to complete installing?

**Comment by jwpeterson** [comment]
I believe what you are seeing is related to a new check that we put in place to ensure that users' environmental `CXXFLAGS` are not in conflict with the build `METHOD` they are using. In this case it looks like you are making a `dbg` build, but there is a `-DNDEBUG` coming from your environment (either `CXXFLAGS` as mentioned or some other type of "flags").
